client/state/data-layer/wpcom-http/actions.js

```diff
@@ -46,6 +46,16 @@ export const http = (
 ) => {
 	const version = apiNamespace ? { apiNamespace } : { apiVersion };
 
+	if ( process.env.NODE_ENV !== 'production' ) {
+		if ( ! ( action || onSuccess ) ) {
+			// eslint-disable-next-line no-console
+			console.warn(
+				'No success handler or default action supplied to the http action. You probably want one or the other. Path: %s',
+				path
+			);
+		}
+	}
```

Comment:
a warning here is appropriate because sometimes we may not even want a success handler. I think the most important thing is informing people when they didn't realize, but not shaming them with an error if they intended as much :smile: